### PR TITLE
MNIST loader for TSNE

### PR DESCRIPTION
**Pull request by MissingNO57**
Reinstate T-SNE Test #824

### CODE REVIEW

File: libs/core/include/core/random/lfg.hpp [hunk 1]
```diff
@@ -73,6 +73,16 @@ class LaggedFibonacciGenerator
     return double(this->operator()()) * inv_double_max_;
   }
 
+  random_type min()
```
Comment (ejfitzgerald) [inline libs/core/include/core/random/lfg.hpp]: I would make `constexpr`

File: libs/core/include/core/random/lfg.hpp [hunk 2]
```diff
@@ -73,6 +73,16 @@ class LaggedFibonacciGenerator
...
+  {
+    return static_cast<random_type>(0);
+  }
+
+  random_type max()
```
Comment (ejfitzgerald) [inline libs/core/include/core/random/lfg.hpp]: Again `constexpr` function here

**Review by ejfitzgerald — DISMISSED** [review]
Just a couple of minor things from my side.

Also, have you tried this build under MacOS? If you need access to the CI Mac then just let me know

**Comment by ejfitzgerald** [comment]
Looks good from my side, will let @baykaner and @pierre-wilmot-fetch-ai do the approval bit 😁

**Review by pierre-wilmot-fetch-ai — APPROVED** [review]

**Pull request merged**